homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -426,19 +579,42 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
 
 
 class XiaomiGenericSensor(XiaomiCoordinatedMiioEntity, SensorEntity):
-    """Representation of a Xiaomi Humidifier sensor."""
-
-    def __init__(self, name, device, entry, unique_id, coordinator, description):
+    """Representation of a Xiaomi generic sensor."""
+
+    def __init__(
+        self,
+        name,
+        device,
+        entry,
+        unique_id,
+        coordinator,
+        description: XiaomiMiioSensorDescription,
+    ):
         """Initialize the entity."""
         super().__init__(name, device, entry, unique_id, coordinator)
-
+        self._attr_device_class = description.device_class
```

Comment:
We don't need to set these entity attributes if we set the entity description attribute, and the description has these corresponding attributes.